### ISSUE TITLE
Keras Classification Final Vbles

### ISSUE THREAD

**Issue by DaniloSerna** [brochero/TFM_DanSer#10]
The classification was made with the same selection of variables as the TMVA:

- j_Deep, "DeepCSV"
- DR_lj, "DeltaR(l-jet)"
- Deta_lj, "DeltaEta(l-jet)"
- DpT_metj, "DeltapT(MET-jet)"
- DPhi_lj, "DeltaPhi(l-jet)"
- Deta_lmet, "Deta_lmet(l-MET)"
- DpT_lj", "DpT_lj(l-jet)"
- DR_metj", "DR_metj"
- DPhi_metj", "DPhi_metj"

With the following linear correlations:

[LinCorrelSig_Kerasv1.pdf](https://github.com/brochero/TFM_DanSer/files/11687035/LinCorrelSig_Kerasv1.pdf)

[LinCorrelBkg_Kerasv1.pdf](https://github.com/brochero/TFM_DanSer/files/11687040/LinCorrelBkg_Kerasv1.pdf)

### **Model result**

The architecture of the sequential model used was:
(9,128,128,128,2)

Compilation used:
- loss='categorical_crossentropy'
- optimizer=Adam(lr=0.01)
- metrics=['accuracy']

Training process:
- NumEpochs=30
- BatchSize=256
- TriesEarlyStopping=3

Discriminant_distribution:

![image](https://github.com/brochero/TFM_DanSer/assets/62732155/48aa20bc-3b93-4881-a956-f1f34e2aa5e9)

ROC curve:
![image](https://github.com/brochero/TFM_DanSer/assets/62732155/127429ba-0d24-4b9d-96e1-feb269eec741)

**Comment by DaniloSerna** [comment]
No Javi, en la nueva distribución `CMSSW_12_0_0` no he podido ni ejecutar Keras:

![image](https://github.com/brochero/TFM_DanSer/assets/62732155/25f722e5-c0c3-41db-9cb3-5d983b306f59)

Quería intentar crear los pesos de PyKeras desde esa versión (por probar algo distinto ya que nada funciona), pero no sé ya cuál es el problema, leí que cambiando la versión de _Keras_ y _Tensorflow_ a la `2.2.0` pero no me da tampoco...

Ahí voy a subir todo esto al repo.

**Comment by DaniloSerna** [comment]
Ya logré ejecutar el `ClassificationKeras.py` en `CMSSW_12_0_0`, para esto toca tener en cuenta:

- No borrar el `tensorflow ` de las librerías importadas
- Ejecutar el script con `python3 ` `ClassificationKeras.py` 
- Lo más importante para evitar los errores de `tensorflow_backend` hay que añadir `tf.keras=True` en el `BookMethod`:

![image](https://github.com/brochero/TFM_DanSer/assets/62732155/9fb821dd-94a7-42e3-8cd8-714aabd174b6)

Ya con estos `weights ` generados (ya hago el respectivo push) aquí se pueden leer en el `BDTValidation `con más optimismos de que en el algún momento logre funcionar.
